### PR TITLE
onchain-oracle

### PR DESCRIPTION
**Pull request by HemendraSharma04**

### CODE REVIEW

File: contracts/protocols/mainnet/onchain-oracle/main.sol [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+import "./interfaces.sol";
+import "./helper.sol";
+
+
+contract Resolver is Helper {
+uint256 ans;
+IERC20[] list;
+
+
+function getRateInUsdc(IERC20[] memory srctokens) public view returns(uint256[] memory){
```
Comment (Vaibhav20201) [inline contracts/protocols/mainnet/onchain-oracle/main.sol]: take address list as parameter

File: contracts/protocols/mainnet/onchain-oracle/main.sol [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+function getConnectors(Ofchain token) public view returns(IERC20[] memory) {
+
+    return token.connectors();
+
+}
+
+
+
+function getWrappedTokens(IERC20 wtoken,bool useWrapper) public view returns(IERC20[] memory){
```
Comment (Vaibhav20201) [inline contracts/protocols/mainnet/onchain-oracle/main.sol]: Whats this function? What does it do?

File: contracts/protocols/mainnet/onchain-oracle/main.sol [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+    Multiwrapper token = getMainnetMultiwrapper();
+
+    (IERC20[] memory wrappedTokens , uint256[] memory rates) =  token.getWrappedTokens(wtoken);
+
+    return wrappedTokens;
+}
+
+
+function oracles() public view returns(IOracle[] memory allOracles) {
```
Comment (Vaibhav20201) [inline contracts/protocols/mainnet/onchain-oracle/main.sol]: What does this function do?

File: contracts/protocols/mainnet/onchain-oracle/interfaces.sol [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+}
+
+interface IOracle {
+    function getRate(IERC20 srcToken, IERC20 dstToken, IERC20 connector) external view returns (uint256 rate, uint256 weight);
+}
+
+
+
+
+interface Ofchain {
```
Comment (Vaibhav20201) [inline contracts/protocols/mainnet/onchain-oracle/interfaces.sol]: name this 1inchOracleInterface

File: contracts/protocols/mainnet/onchain-oracle/main.sol [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+
+contract Resolver is Helper {
+uint256 ans;
+IERC20[] list;
+
+
+function getRateInUsdc(IERC20[] memory srctokens) public view returns(uint256[] memory){
+    uint256[] memory prices = new uint256[](srctokens.length);
+    Ofchain token  = getMainnetoracle();
```
Comment (Vaibhav20201) [inline contracts/protocols/mainnet/onchain-oracle/main.sol]: Define the oracle as a storage/state variable and use it everywhere

File: contracts/protocols/mainnet/onchain-oracle/interfaces.sol [hunk 6]
```diff
@@ -0,0 +1,54 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.0;
+
+ enum OracleType { WETH, ETH, WETH_ETH }
+
+interface IERC20 {
```
Comment (Vaibhav20201) [inline contracts/protocols/mainnet/onchain-oracle/interfaces.sol]: Use import IERC20 interface directly for openzepplin's library

File: contracts/protocols/mainnet/onchain-oracle/main.sol [hunk 7]
```diff
@@ -0,0 +1,50 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.0;
+
+import "./interfaces.sol";
+import "./helper.sol";
+
+
+contract Resolver is Helper {
+uint256 ans;
```
Comment (Vaibhav20201) [inline contracts/protocols/mainnet/onchain-oracle/main.sol]: Indent all variables and functions properly

File: contracts/protocols/mainnet/onchain-oracle/helper.sol [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+        return Ofchain(0x07D91f5fb9Bf7798734C3f606dB065549F6893bb);
+    }
+
+function getMainnetMultiwrapper() public pure returns(Multiwrapper){
+    return Multiwrapper(0x931e32B6D112F7BE74B16f7FBc77D491B30fe18c);
+}
+
+function getUsdcToken() public pure returns(IERC20){
+    return IERC20(0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48);
+}
```
Comment (Vaibhav20201) [inline contracts/protocols/mainnet/onchain-oracle/helper.sol]: Define all these as state/storage variables in the core contract [here](https://github.com/Instadapp/dsa-resolvers/pull/54#discussion_r802037564) and use it. No need of these functions.

File: contracts/protocols/mainnet/onchain-oracle/main.sol [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+IERC20[] list;
+
```
Comment (Vaibhav20201) [inline contracts/protocols/mainnet/onchain-oracle/main.sol]: here

**Review by Vaibhav20201 — CHANGES_REQUESTED** [review]

**Comment by thrilok209** [comment]
@Vaibhav20201 @CodeNinja04 What's the status on this? Ready to deploy and merge it?

**Comment by thrilok209** [comment]
Closing this for now. Can revisit whenever it's required.

**Pull request closed without merging**